### PR TITLE
Nbelakovski/serialize vehicles and wheels

### PR DESCRIPTION
**Pull request by nbelakovski**
Added serialization for vehicles (by adding serialization for action interfaces), and then, in a separate commit, for wheels associated with vehicles.

I tested by making modifications to one of the examples in the example browser and everything worked as expected. I can provide some details of that testing if you like.

Mentioning @dbacchet since he is also interested in this.

### CODE REVIEW

**Comment by nbelakovski** [comment]
It seems the compiler didn't like some of the C++ 11 features I was using. Are there any plans for bullet to support C++11? I tried searching the forums, but the search function didn't like the search term "C++".

**Comment by erwincoumans** [comment]
No, coding style guide is here:
https://docs.google.com/document/d/1u9vyzPtrVoVhYqQOGNWUgjRbfwfCdIts_NzmvgiJ144/edit

**Comment by erwincoumans** [comment]
no auto, no dynamic_cast or RTTI etc.

instead of
~~~~
chassis = dynamic_cast<btRigidBody*>(*obj);
~~~~
use
chassis = btRigidBody::upcast(colObj);
and check if chassis is non-zero.

File: src/BulletDynamics/Vehicle/btRaycastVehicle.cpp [hunk 1]
```diff
@@ -551,46 +561,43 @@ void	btRaycastVehicle::updateFriction(btScalar	timeStep)
 			m_forwardImpulse[i] = btScalar(0.);
 
 		}
-	
+
```
Comment (erwincoumans) [inline src/BulletDynamics/Vehicle/btRaycastVehicle.cpp]: what is happening here? Is this a code change or just formatting? If formatting, please undo.

Comment (nbelakovski) [reply src/BulletDynamics/Vehicle/btRaycastVehicle.cpp]: It's hard to see, but there were some unnecessary curly brackets so I removed them and unindented the rest. It's clearer to see if you add '?w=1' to the url for the PR in order to hide whitespace changes. I didn't really pay attention to this part of the diff as I was preparing it, but if you'd like I can either revert it or go back through and make sure it's just a curly bracket removal and indentation change, and that there are no lines added or removed and no tabs changed to spaces.

**Comment by nbelakovski** [comment]
Will Bullet ever support C++11, and is there a forum discussion on this topic, or other more appropriate place to discuss that question other than this PR?

Apart from style and formatting, do you have any big picture questions/comments/concerns? Is this functionality that you'd like to see added to Bullet, and if so, are you broadly satisfied with the implementation, or do you have an alternative implementation that you'd like to propose?

**Comment by erwincoumans** [comment]
>> Will Bullet ever support C++11
Perhaps in the future indeed, but don't expect it soon.

>>do you have any big picture questions/comments/concerns?
I'll try to look at it at some stage, note that this project lacks resources/time and at the moment I'm  focussing at PyBullet and doing research on the intersection of machine learning and physics simulation.

**Comment by nbelakovski** [comment]
> I'll try to look at it at some stage

Could you be more specific? We have other work planned that we'd like to contribute back, but if we can't expect timely approval on PRs, then we'll have to keep our development internal. And we'd really like to be able to contribute back!

**Comment by nbelakovski** [comment]
Closing for now since it doesn't look like this is on a path to get merged. I will leave the branch as it is in my repo.

**Pull request closed without merging**